From 1fac453f7b6a25724a69c4fd284ba479befcfb24 Mon Sep 17 00:00:00 2001
From: Stas Cymbalov <dummyunit@gmail.com>
Date: Sat, 4 May 2019 23:02:37 +0300
Subject: [PATCH] Add missing headers

* memory.h is needed for memset, without it non-unity build fails on Linux.
* fcntl.h is needed for open, without it non-unity build fails on Linux
  when musl is used.
---
 Code/Core/Process/SystemMutex.cpp                  | 1 +
 Code/Tools/FBuild/FBuildCore/Protocol/Protocol.cpp | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/Code/Core/Process/SystemMutex.cpp b/Code/Core/Process/SystemMutex.cpp
index 79f4713..73614d8 100644
--- a/Code/Core/Process/SystemMutex.cpp
+++ b/Code/Core/Process/SystemMutex.cpp
@@ -14,6 +14,7 @@
 #if defined( __LINUX__ ) || defined( __APPLE__ )
     #include <errno.h>
     #include <sys/file.h>
+    #include <fcntl.h>
     #include <unistd.h>
 #endif
 
diff --git a/Code/Tools/FBuild/FBuildCore/Protocol/Protocol.cpp b/Code/Tools/FBuild/FBuildCore/Protocol/Protocol.cpp
index 46eadec..69d5cfa 100644
--- a/Code/Tools/FBuild/FBuildCore/Protocol/Protocol.cpp
+++ b/Code/Tools/FBuild/FBuildCore/Protocol/Protocol.cpp
@@ -10,6 +10,8 @@
 #include "Core/FileIO/MemoryStream.h"
 #include "Core/Network/TCPConnectionPool.h"
 
+// system
+#include <memory.h> // for memset
 #if defined( __APPLE__ ) || defined( __LINUX__ )
     #include <unistd.h> // for ::gethostname
 #endif
-- 
2.21.0

